fusion-cli/build/plugins/instrumented-import-dependency-template-plugin.js

```diff
@@ -48,80 +99,107 @@ class InstrumentedImportDependencyTemplate extends ImportDependencyTemplate {
+      translationKeys = instrumentation.translationKeys;
+    } else if (this.clientChunkIndex) {
+      // Template invoked without InstrumentedImportDependency
       // server-side, use values from client bundle
       let ids = this.clientChunkIndex.get(
         (dep.module && dep.module.resource) || dep.originModule.resource
       );
       chunkIds = ids ? Array.from(ids) : [];
     } else {
-      // client-side, use built-in values
-      chunkIds = getChunkGroupIds(depBlock.chunkGroup);
+      chunkIds = getChunkGroupIds(dep.block.chunkGroup);
     }
 
+    // This is a hack. Some production builds had undefined for the module.id
+    // which broke the build.
+    dep.module.id =
+      dep.module.id || (dep.module.identifier && dep.module.identifier());
+
+    const content = runtime.moduleNamespacePromise({
+      block: dep.block,
+      module: dep.module,
+      request: dep.request,
+      strict: dep.originModule.buildMeta.strictHarmonyModule,
+      message: 'import()',
+    });
+
     // Add the following properties to the promise returned by import()
     // - `__CHUNK_IDS`: the webpack chunk ids for the dynamic import
     // - `__MODULE_ID`: the webpack module id of the dynamically imported module. Equivalent to require.resolveWeak(path)
+    // - `__I18N_KEYS`: the translation keys used in the client chunk group for this import()
     const customContent = chunkIds
       ? `Object.defineProperties(${content}, {
         "__CHUNK_IDS": {value:${JSON.stringify(chunkIds)}},
-        "__MODULE_ID": {value:${JSON.stringify(dep.module.id)}}
+        "__MODULE_ID": {value:${JSON.stringify(dep.module.id)}},
+        "__I18N_KEYS": {value:${JSON.stringify(translationKeys)}}
         })`
       : content;
 
     // replace with `customContent` instead of `content`
     source.replace(depBlock.range[0], depBlock.range[1] - 1, customContent);
   }
-}
+};
 
 /**
  * Webpack plugin to replace standard ImportDependencyTemplate with custom one
  * See InstrumentedImportDependencyTemplate for more info
  */
-
 class InstrumentedImportDependencyTemplatePlugin {
-  /*:: clientChunkIndexState: ?ClientChunkMetadataState; */
+  /*:: opts: InstrumentationPluginOpts;*/
 
-  constructor(clientChunkIndexState /*: ?ClientChunkMetadataState*/) {
-    this.clientChunkIndexState = clientChunkIndexState;
+  constructor(opts /*: InstrumentationPluginOpts*/) {
+    this.opts = opts;
   }
 
   apply(compiler /*: any */) {
     const name = this.constructor.name;
-    /**
-     * The standard plugin is added on `compile`,
-     * which sets the default value for `ImportDependency` in  the `dependencyTemplates` map.
-     * `make` is the subsequent lifeycle method, so we can override this value here.
-     */
-    compiler.hooks.make.tapAsync(name, (compilation, done) => {
-      if (this.clientChunkIndexState) {
-        // server
-        this.clientChunkIndexState.result.then(chunkIndex => {
+    if (this.opts.compilation === 'server') {
```

Comment:
I ended up keeping the server compilation as it was before essentially. This promise makes it a little weird to work with, so I've kept the client and server compilations separate.